src/main/java/org/jabref/logic/importer/fetcher/CiteSeer.java

```diff
@@ -45,32 +48,109 @@ public Optional<HelpFile> getHelpPage() {
         return Optional.of(HelpFile.FETCHER_CITESEERX);
     }
 
+    @Override
+    public List<BibEntry> performSearch(QueryNode luceneQuery) throws FetcherException {
+        // ADR-0014
+        URL urlForQuery;
+        try {
+            urlForQuery = getURLForQuery(luceneQuery);
+        } catch (URISyntaxException | MalformedURLException | FetcherException e) {
+            throw new FetcherException("Search URI crafted from complex search query is malformed", e);
+        }
+
+        String payload = getPayloadString(new DefaultQueryTransformer().transformLuceneQuery(luceneQuery).orElse(""));
+        return getBibEntries(urlForQuery, payload);
+    }
+
+    private List<BibEntry> getBibEntries(URL urlForQuery, String payload) throws FetcherException {
+        try {
+            HttpURLConnection httpConn = (HttpURLConnection) urlForQuery.openConnection();
```

Comment:
Insteadd of working on such a low level with httpConn, I would propose you take a look at Unirest
http://kong.github.io/unirest-java/